app/models/post.rb

```diff
@@ -1,2 +1,15 @@
 class Post < ApplicationRecord
+  # Relationship
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, class_name: 'Comment'
+  has_many :likes, class_name: 'Like'
+
+  def recent_comments
+    comments.order(created_at: :asc).limit(5)
+  end
+
+  def update_post_count
+    user.update(post_counter: user.posts.count)
+  end
+  4
 end
```

Comment:
- [ ] Good job adding method for the Post model 👏 . I will suggest you to use `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private, It will optimize the work 👍 

 ```ruby
...
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
...
```